### PR TITLE
_Bool fix for cpp compilation

### PR DESCRIPTION
**Pull request by jordo**
_Bool isn't defined when compiling from (.mm) objc++ files.

### CODE REVIEW

**Comment by davidchisnall** [comment]
What is the `#include <stdbool.h>` for?  This defines `bool` in C and should have no effect in C++ mode.  I think it would be cleaner to either `#include <stdbool.h>` in C mode and just use `bool` everywhere (may break old C code, but is probably fine) or define an `OBJC_BOOL` macro that is either `bool` or `_Bool` depending on whether we're in C or C++ mode.

**Comment by jordo** [comment]
oops  I thought i have taken that out before the pull req, I can remove.  Just using bool does break C code in other libraries (dispatch) for example.    But ya, agreed OBJC_BOOL would be cleaner and I can guard it, as I wasn't sure if OBJC_BOOL could be defined on some other lib or something.

File: objc/blocks_runtime.h [hunk 1]
```diff
@@ -2,35 +2,28 @@
...
 
+
 /*
  * Blocks Runtime
  */
 #include "Availability.h"
 #ifdef __cplusplus
-#include <stdbool.h>
+#define OBJC_BOOL_TYPE bool
+#define OBJC_BLOCK_PTR_TYPE void*
```
Comment (davidchisnall) [inline objc/blocks_runtime.h]: Is this consistent with Apple?  I'm not sure why we ended up with `void*` in Objective-C++ and `id` in Objective-C.

Comment (jordo) [reply objc/blocks_runtime.h]: OK.  maybe i'll refactor to be as consistent with apple as possible (they seem to include #include <stdbool.h> and just use bool as return type as well).  That may be best moving forward for compatibility with their header.

Comment (jordo) [reply objc/blocks_runtime.h]: OK, should look better now.

**Pull request merged**